### PR TITLE
chore(sqlsmith): reduce out of range occurrence

### LINKED ISSUE #7818: false positive by sqlsmith: function `split_part`
### Describe the bug

-- Setup
<details>

```sql
CREATE TABLE supplier (s_suppkey INT, s_name CHARACTER VARYING, s_address CHARACTER VARYING, s_nationkey INT, s_phone CHARACTER VARYING, s_acctbal NUMERIC, s_comment CHARACTER VARYING, PRIMARY KEY (s_suppkey));CREATE TABLE part (p_partkey INT, p_name CHARACTER VARYING, p_mfgr CHARACTER VARYING, p_brand CHARACTER VARYING, p_type CHARACTER VARYING, p_size INT, p_container CHARACTER VARYING, p_retailprice NUMERIC, p_comment CHARACTER VARYING, PRIMARY KEY (p_partkey));CREATE TABLE partsupp (ps_partkey INT, ps_suppkey INT, ps_availqty INT, ps_supplycost NUMERIC, ps_comment CHARACTER VARYING, PRIMARY KEY (ps_partkey, ps_suppkey));CREATE TABLE customer (c_custkey INT, c_name CHARACTER VARYING, c_address CHARACTER VARYING, c_nationkey INT, c_phone CHARACTER VARYING, c_acctbal NUMERIC, c_mktsegment CHARACTER VARYING, c_comment CHARACTER VARYING, PRIMARY KEY (c_custkey));CREATE TABLE orders (o_orderkey BIGINT, o_custkey INT, o_orderstatus CHARACTER VARYING, o_totalprice NUMERIC, o_orderdate DATE, o_orderpriority CHARACTER VARYING, o_clerk CHARACTER VARYING, o_shippriority INT, o_comment CHARACTER VARYING, PRIMARY KEY (o_orderkey));CREATE TABLE lineitem (l_orderkey BIGINT, l_partkey INT, l_suppkey INT, l_linenumber INT, l_quantity NUMERIC, l_extendedprice NUMERIC, l_discount NUMERIC, l_tax NUMERIC, l_returnflag CHARACTER VARYING, l_linestatus CHARACTER VARYING, l_shipdate DATE, l_commitdate DATE, l_receiptdate DATE, l_shipinstruct CHARACTER VARYING, l_shipmode CHARACTER VARYING, l_comment CHARACTER VARYING, PRIMARY KEY (l_orderkey, l_linenumber));CREATE TABLE nation (n_nationkey INT, n_name CHARACTER VARYING, n_regionkey INT, n_comment CHARACTER VARYING, PRIMARY KEY (n_nationkey));CREATE TABLE region (r_regionkey INT, r_name CHARACTER VARYING, r_comment CHARACTER VARYING, PRIMARY KEY (r_regionkey));CREATE TABLE person (id BIGINT, name CHARACTER VARYING, email_address CHARACTER VARYING, credit_card CHARACTER VARYING, city CHARACTER VARYING, state CHARACTER VARYING, date_time TIMESTAMP, extra CHARACTER VARYING, PRIMARY KEY (id));CREATE TABLE auction (id BIGINT, item_name CHARACTER VARYING, description CHARACTER VARYING, initial_bid BIGINT, reserve BIGINT, date_time TIMESTAMP, expires TIMESTAMP, seller BIGINT, category BIGINT, extra CHARACTER VARYING, PRIMARY KEY (id));CREATE TABLE bid (auction BIGINT, bidder BIGINT, price BIGINT, channel CHARACTER VARYING, url CHARACTER VARYING, date_time TIMESTAMP, extra CHARACTER VARYING);CREATE TABLE alltypes1 (c1 BOOLEAN, c2 SMALLINT, c3 INT, c4 BIGINT, c5 REAL, c6 DOUBLE, c7 NUMERIC, c8 DATE, c9 CHARACTER VARYING, c10 TIME, c11 TIMESTAMP, c13 INTERVAL, c14 STRUCT<a INT>, c15 INT[], c16 CHARACTER VARYING[]);CREATE TABLE alltypes2 (c1 BOOLEAN, c2 SMALLINT, c3 INT, c4 BIGINT, c5 REAL, c6 DOUBLE, c7 NUMERIC, c8 DATE, c9 CHARACTER VARYING, c10 TIME, c11 TIMESTAMP, c13 INTERVAL, c14 STRUCT<a INT>, c15 INT[], c16 CHARACTER VARYING[]);CREATE MATERIALIZED VIEW m0 AS SELECT (-1168308523) AS col_0, ((SMALLINT '32767') & (SMALLINT '-32768')) AS col_1 FROM auction AS t_0 JOIN person AS t_1 ON t_0.item_name = t_1.extra WHERE false GROUP BY t_0.id, t_0.item_name, t_1.extra, t_1.state, t_1.credit_card;CREATE MATERIALIZED VIEW m1 AS SELECT (REAL '-360864233') AS col_0, (REAL '967633323') AS col_1, (634625719) AS col_2, (((SMALLINT '5328') # (SMALLINT '1')) >> (INT '604075833')) AS col_3 FROM bid AS t_0 WHERE false GROUP BY t_0.channel HAVING false;CREATE MATERIALIZED VIEW m2 AS SELECT t_0.c13 AS col_0, t_0.c8 AS col_1, (t_0.c6 / ((REAL '887594576') * (REAL '1516617131'))) AS col_2, false AS col_3 FROM alltypes1 AS t_0 WHERE t_0.c1 GROUP BY t_0.c13, t_0.c10, t_0.c9, t_0.c11, t_0.c7, t_0.c8, t_0.c14, t_0.c6;CREATE MATERIALIZED VIEW m3 AS SELECT TIME '23:51:00' AS col_0, TIMESTAMP '2023-02-08 14:06:51' AS col_1 FROM auction AS t_0 WHERE CAST((char_length(t_0.item_name)) AS BOOLEAN) GROUP BY t_0.reserve, t_0.id, t_0.date_time;CREATE MATERIALIZED VIEW m4 AS SELECT DATE '2023-02-09' AS col_0 FROM partsupp AS t_0 GROUP BY t_0.ps_supplycost, t_0.ps_availqty, t_0.ps_suppkey;CREATE MATERIALIZED VIEW m5 AS SELECT (INTERVAL '1') AS col_0, (7041590175945126020) AS col_1, 'nrerASRZxN' AS col_2, (1936772772) AS col_3 FROM alltypes2 AS t_0 GROUP BY t_0.c10, t_0.c2, t_0.c4 HAVING true;CREATE MATERIALIZED VIEW m6 AS SELECT (0) AS col_0, (2179484222877753951) AS col_1, ((INTERVAL '0') + (coalesce(NULL, NULL, NULL, NULL, TIME '14:06:52', NULL, NULL, NULL, NULL, NULL))) AS col_2, DATE '2023-02-09' AS col_3 FROM region AS t_0 GROUP BY t_0.r_name, t_0.r_comment;CREATE MATERIALIZED VIEW m7 AS SELECT (-9223372036854775808) AS col_0 FROM m4 AS t_0 GROUP BY t_0.col_0;CREATE MATERIALIZED VIEW m8 AS SELECT (INT '1117551957') AS col_0, (FLOAT '0') AS col_1, (FLOAT '1879219110') AS col_2, TIMESTAMP '2023-02-09 13:06:52' AS col_3 FROM m1 AS t_0 GROUP BY t_0.col_0, t_0.col_3 HAVING true;CREATE MATERIALIZED VIEW m9 AS SELECT (FLOAT '-2147483648') AS col_0 FROM supplier AS t_0 WHERE true GROUP BY t_0.s_name, t_0.s_suppkey, t_0.s_nationkey;
```

</details>

-- Query

```sql
CREATE MATERIALIZED VIEW stream_query AS SELECT min((INT '-905002677')) FILTER(WHERE ('LgD8RpfO1p') IN ('JH16vflD9D', 'W1bYtYIsZ3', (replace('H2PCuTwF5m', 'ww36yfhxns', 'v8pegJ9ceo')), 'g8h6wVVy1i', 'blhXhaWw5J', 'mgK5KWkOAa', (split_part('Jgan3InunH', 'KAUTxHnNGz', ((INT '0')))))) AS col_0 FROM auction AS t_0 GROUP BY t_0.reserve, t_0.category, t_0.seller, t_0.id, t_0.description;
```

---- END

Reason:

```
db error: ERROR: ExecuteError: internal error: Rpc error: gRPC error (Internal error): Expression error: Invalid parameter data: can't be zero
', /risingwave/src/tests/sqlsmith/src/runner.rs:197:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:64:14
   2: risingwave_sqlsmith::runner::validate_response
             at ./src/tests/sqlsmith/src/runner.rs:197:13
   3: risingwave_sqlsmith::runner::test_stream_queries::{{closure}}
             at ./src/tests/sqlsmith/src/runner.rs:105:20
   4: risingwave_sqlsmith::runner::run::{{closure}}
             at ./src/tests/sqlsmith/src/runner.rs:32:77
   5: sqlsmith::main::{{closure}}
             at ./src/tests/sqlsmith/src/bin/main.rs:98:43
   6: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-0.1.37/src/instrument.rs:272:9
   7: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/park.rs:283:63
   8: tokio::runtime::coop::with_budget
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/coop.rs:102:5
   9: tokio::runtime::coop::budget
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/coop.rs:68:5
  10: tokio::runtime::park::CachedParkThread::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/park.rs:283:31
  11: tokio::runtime::context::BlockingRegionGuard::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/context.rs:315:13
  12: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/scheduler/multi_thread/mod.rs:66:9
  13: tokio::runtime::runtime::Runtime::block_on
             at ./.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.24.2/src/runtime/runtime.rs:284:45
  14: sqlsmith::main
             at ./src/tests/sqlsmith/src/bin/main.rs:98:5
  15: core::ops::function::FnOnce::call_once
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/ops/function.rs:250:5
```

### To Reproduce

_No response_

### Expected behavior

The error seems to be expected:

```
dev=> SELECT SPLIT_PART('A,B,C', ',', 2);
 split_part 
------------
 B
(1 row)

dev=> SELECT SPLIT_PART('A,B,C', ',', 0);
ERROR:  QueryError: Expr error: Invalid parameter data: can't be zero
```

PG:
```
xxchan@/tmp:postgres> SELECT SPLIT_PART('A,B,C', ',', 2);
+------------+
| split_part |
|------------|
| B          |
+------------+
SELECT 1
Time: 0.010s
xxchan@/tmp:postgres> SELECT SPLIT_PART('A,B,C', ',', 0);
field position must not be zero
Time: 0.006s
xxchan@/tmp:postgres>
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve errors in https://github.com/risingwavelabs/risingwave/pull/7761.

fix #7818

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by kwannoel** [comment]
Btw I've tried it on #7761, there are still some errors, which need some investigation:
```
---- END

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: CapacityError: insufficient capacity', /Users/noelkwan/.cargo/registry/src/github.com-1ecc6299db9ec823/arrayvec-0.7.2/src/arrayvec_impl.rs:39:32
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:64:14
   2: core::result::unwrap_failed
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/result.rs:1790:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/result.rs:1112:23
   4: arrayvec::arrayvec_impl::ArrayVecImpl::push
             at /Users/noelkwan/.cargo/registry/src/github.com-1ecc6299db9ec823/arrayvec-0.7.2/src/arrayvec_impl.rs:39:9
   5: arrayvec::arrayvec::ArrayVec<T,_>::push
             at /Users/noelkwan/.cargo/registry/src/github.com-1ecc6299db9ec823/arrayvec-0.7.2/src/arrayvec.rs:176:9
   6: rust_decimal::str::to_str_internal
             at /Users/noelkwan/.cargo/registry/src/github.com-1ecc6299db9ec823/rust_decimal-1.27.0/src/str.rs:30:9
   7: <rust_decimal::decimal::Decimal as core::fmt::Display>::fmt
             at /Users/noelkwan/.cargo/registry/src/github.com-1ecc6299db9ec823/rust_decimal-1.27.0/src/decimal.rs:2314:33
   8: <&T as core::fmt::Display>::fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:2377:62
   9: core::fmt::write
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:1213:17
  10: core::fmt::Formatter::write_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:1667:9
  11: <risingwave_common::types::decimal::Decimal as core::fmt::Display>::fmt
             at /Users/noelkwan/projects/risingwave/src/common/src/types/decimal.rs:33:23
  12: <&T as core::fmt::Display>::fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:2377:62
  13: core::fmt::write
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:1213:17
  14: core::fmt::Write::write_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/mod.rs:197:9
  15: <risingwave_common::types::decimal::Decimal as risingwave_common::types::to_text::ToText>::write
             at /Users/noelkwan/projects/risingwave/src/common/src/types/decimal.rs:47:9
  16: <risingwave_common::types::ScalarRefImpl as risingwave_common::types::to_text::ToText>::write
             at /Users/noelkwan/projects/risingwave/src/common/src/types/to_text.rs:223:58
  17: risingwave_common::types::to_text::ToText::to_text
             at /Users/noelkwan/projects/risingwave/src/common/src/types/to_text.rs:61:9
```

`ddl + query`
```
---- START

-- Failing SQL setup code:
CREATE TABLE supplier (
    s_suppkey INTEGER,
    s_name VARCHAR,
    s_address VARCHAR,
    s_nationkey INTEGER,
    s_phone VARCHAR,
    s_acctbal NUMERIC,
    s_comment VARCHAR,
    PRIMARY KEY (s_suppkey)
);

CREATE TABLE part (
    p_partkey INTEGER,
    p_name VARCHAR,
    p_mfgr VARCHAR,
    p_brand VARCHAR,
    p_type VARCHAR,
    p_size INTEGER,
    p_container VARCHAR,
    p_retailprice NUMERIC,
    p_comment VARCHAR,
    PRIMARY KEY (p_partkey)
);

CREATE TABLE partsupp (
    ps_partkey INTEGER,
    ps_suppkey INTEGER,
    ps_availqty INTEGER,
    ps_supplycost NUMERIC,
    ps_comment VARCHAR,
    PRIMARY KEY (ps_partkey, ps_suppkey)
);

CREATE TABLE customer (
    c_custkey INTEGER,
    c_name VARCHAR,
    c_address VARCHAR,
    c_nationkey INTEGER,
    c_phone VARCHAR,
    c_acctbal NUMERIC,
    c_mktsegment VARCHAR,
    c_comment VARCHAR,
    PRIMARY KEY (c_custkey)
);

CREATE TABLE orders (
    o_orderkey BIGINT,
    o_custkey INTEGER,
    o_orderstatus VARCHAR,
    o_totalprice NUMERIC,
    o_orderdate DATE,
    o_orderpriority VARCHAR,
    o_clerk VARCHAR,
    o_shippriority INTEGER,
    o_comment VARCHAR,
    PRIMARY KEY (o_orderkey)
);

CREATE TABLE lineitem (
    l_orderkey BIGINT,
    l_partkey INTEGER,
    l_suppkey INTEGER,
    l_linenumber INTEGER,
    l_quantity NUMERIC,
    l_extendedprice NUMERIC,
    l_discount NUMERIC,
    l_tax NUMERIC,
    l_returnflag VARCHAR,
    l_linestatus VARCHAR,
    l_shipdate DATE,
    l_commitdate DATE,
    l_receiptdate DATE,
    l_shipinstruct VARCHAR,
    l_shipmode VARCHAR,
    l_comment VARCHAR,
    PRIMARY KEY (l_orderkey, l_linenumber)
);

CREATE TABLE nation (
    n_nationkey INTEGER,
    n_name VARCHAR,
    n_regionkey INTEGER,
    n_comment VARCHAR,
    PRIMARY KEY (n_nationkey)
);

CREATE TABLE region (
    r_regionkey INTEGER,
    r_name VARCHAR,
    r_comment VARCHAR,
    PRIMARY KEY (r_regionkey)
);
CREATE TABLE person (
    id BIGINT,
    name VARCHAR,
    email_address VARCHAR,
    credit_card VARCHAR,
    city VARCHAR,
    state VARCHAR,
    date_time TIMESTAMP,
    extra VARCHAR,
    PRIMARY KEY (id)
);

CREATE TABLE auction (
    id BIGINT,
    item_name VARCHAR,
    description VARCHAR,
    initial_bid BIGINT,
    reserve BIGINT,
    date_time TIMESTAMP,
    expires TIMESTAMP,
    seller BIGINT,
    category BIGINT,
    extra VARCHAR,
    PRIMARY KEY (id)
);

CREATE TABLE bid (
    auction BIGINT,
    bidder BIGINT,
    price BIGINT,
    channel VARCHAR,
    url VARCHAR,
    date_time TIMESTAMP,
    extra VARCHAR
);
CREATE TABLE alltypes1 (
    c1 boolean,
    c2 smallint,
    c3 integer,
    c4 bigint,
    c5 real,
    c6 double precision,
    c7 numeric,
    c8 date,
    c9 varchar,
    c10 time without time zone,
    c11 timestamp without time zone,
    -- TODO(Noel): Many timestamptz expressions are unsupported in RW, leave this out for now.
    -- c12 timestamp with time zone,
    c13 interval,
    c14 struct < a integer >,
    c15 integer [],
    c16 varchar []
);

CREATE TABLE alltypes2 (
    c1 boolean,
    c2 smallint,
    c3 integer,
    c4 bigint,
    c5 real,
    c6 double precision,
    c7 numeric,
    c8 date,
    c9 varchar,
    c10 time without time zone,
    c11 timestamp without time zone,
    -- TODO(Noel): Many timestamptz expressions are unsupported in RW, leave this out for now.
    -- c12 timestamp with time zone,
    c13 interval,
    c14 struct < a integer >,
    c15 integer [],
    c16 varchar []
);CREATE MATERIALIZED VIEW m0 AS SELECT (REAL '817') AS col_0, (- (hop_0.c2 | ((SMALLINT '433') >> hop_0.c2))) AS col_1, avg(hop_0.c6) FILTER(WHERE CAST((INT '2147483647') AS BOOLEAN)) AS col_2 FROM hop(alltypes1, alltypes1.c11, INTERVAL '60', INTERVAL '36288000') AS hop_0 GROUP BY hop_0.c8, hop_0.c7, hop_0.c6, hop_0.c13, hop_0.c2, hop_0.c15;CREATE MATERIALIZED VIEW m1 AS SELECT (271) AS col_0, t_0.ps_supplycost AS col_1 FROM partsupp AS t_0 WHERE false GROUP BY t_0.ps_supplycost HAVING CAST((INT '966') AS BOOLEAN);CREATE MATERIALIZED VIEW m2 AS SELECT t_0.o_orderkey AS col_0, t_0.o_orderkey AS col_1, (upper('YP067GJfX1')) AS col_2 FROM orders AS t_0 GROUP BY t_0.o_orderkey, t_0.o_totalprice, t_0.o_orderpriority, t_0.o_comment;

-- Failing SQL query:
CREATE MATERIALIZED VIEW m3 AS SELECT (308) AS col_0, (936) AS col_1, (SMALLINT '66') AS col_2, (CAST(NULL AS STRUCT<a TIMESTAMP, b NUMERIC, c CHARACTER VARYING>)) AS col_3 FROM tumble(person, person.date_time, INTERVAL '3600') AS tumble_0 GROUP BY tumble_0.credit_card, tumble_0.name, tumble_0.state;


```

**Comment by jon-chuang** [comment]
It appears to be a large decimal. Let's figure out why.

**Comment by jon-chuang** [comment]
Filed a bug report: https://github.com/paupino/rust-decimal/issues/574

**Comment by jon-chuang** [comment]
Actually, the failing value should be impossible, as the scale should never exceed 28. This seems to be a bug within risingwave, we may have ended up with a scale that is too large.

**Comment by jon-chuang** [comment]
Actually, this is a rust_decimal bug.

File: src/tests/sqlsmith/src/validation.rs [hunk 1]
```diff
@@ -15,8 +15,9 @@
 //! Provides validation logic for expected errors.
 use risingwave_expr::ExprError;
 
-fn is_division_by_zero_err(db_error: &str) -> bool {
-    db_error.contains(&ExprError::DivisionByZero.to_string())
+/// Ignore errors related to `0`.
+fn is_zero_err(db_error: &str) -> bool {
+    db_error.contains(&ExprError::DivisionByZero.to_string()) || db_error.contains("can't be zero")
```
Comment (xiangjinwu) [inline src/tests/sqlsmith/src/validation.rs:20]: Guess soon we will need to ignore the following errors as well:
```
test=# select pow(0, -1);
ERROR:  zero raised to a negative power is undefined
test=# select pow(-1, 0.5);
ERROR:  a negative number raised to a non-integer power yields a complex result
```

Right now our behavior is inconsistent with PG, with the former returning `Numeric out of range` (already handled) and latter evaluates to `NaN` (no error).

This suggest we may need to categorize certain errors as "safe to ignore in sqlsmith" proactively. But I cannot see clearly where the boundary is. Or maybe the current approach of adding new ones reactively is good enough? Just raising the discussion and would like to hear what your thoughts.

File: src/tests/sqlsmith/src/validation.rs [hunk 2]
```diff
@@ -71,7 +72,7 @@ fn is_neg_substr_error(db_error: &str) -> bool {
 /// 2. These errors seldom occur, skipping them won't affect overall effectiveness of sqlsmith.
```
Comment (kwannoel) [inline src/tests/sqlsmith/src/validation.rs:72]: Not such a good criteria.

File: src/tests/sqlsmith/src/validation.rs (hunk 1 above)
Comment (kwannoel) [reply src/tests/sqlsmith/src/validation.rs:20]: Some preliminary thoughts:
There are several criteria which I consider when ignoring errors:
1. Is it error caused by user? e.g. `DivisionByZeroError`
1. Is the error message expected?
1. Is it complex to workaround it?

If yes to all 3, can consider including it in ignored errors.

I think the currently set of errors ignored by sqlsmith includes most if not all the categories which are safe to ignore.

What is done reactively is including errors in these existing categories.
We can add more errors proactively to these categories, I suspect several errors are not ignored:
- zero errors
- out of range errors
- negative errors

Criteria 3 is particularly important for sqlsmith. My thinking is that it is worth the trade-off: 
1. Pro: Keep sqlsmith simple, easy to maintain.
2. Tradeoff: Perhaps some errors should not be returned, but these are rare.

I think there's still some room for improvements which are lower effort but higher reward: see #3896 .

Eventually we can do something like:
- Generate only non erroring arguments for functions.
- Workaround unsupported features (`nested_loop_join`, `subquery_unnesting`).

Comment (kwannoel) [reply src/tests/sqlsmith/src/validation.rs:20]: actually they occur more frequently than expected. We should generate queries which do not contain these errors. #7866

**Review by jon-chuang — APPROVED** [review]
LGTM

Let's treat this as a provisional fix. Generally I'm in agreement, we should limit the generation so that we have fewer false positives.

**Pull request merged**